pkg/autodiscovery/integration/config.go

```diff
@@ -58,6 +58,11 @@ type CommonInstanceConfig struct {
 	EmptyDefaultHostname  bool `yaml:"empty_default_hostname"`
 }
 
+// ExtraInstanceConfig holds the reserved fields for the extra instance id 
+type ExtraInstanceConfig struct {
```

Comment:
What's your rationale behind creating a dedicated type instead of extending `CommonInstanceConfig`

Reply:
Since it is extra, it doesn't seem Common to me.
Also, we could add several other fields that could be used as this extra-id.
What are your thoughts?

Reply:
i think this could be an optional field in the common instance config ?

Reply:
i'm also not a fan of `ExtraID`, It's name doesn't tell much about what purpose it serves. What about `HumanFriendlyID` or something in that vein?

Reply:
also, it's strange to make it a generic name, and then to hardcode its yaml key to `namespace`. If you make it part of `CommonInstanceConfig` you can add logic for parsing it in [CommonConfigure](https://github.com/DataDog/datadog-agent/blob/11004e8206d9450ef37f1b7d80df23e249ebb05b/pkg/collector/corechecks/checkbase.go#L65) instead of the new `GetExtraIDForInstance `. We could look for `namespace`, and maybe fallback to `human_friendly_id` if `namespace` is not there.
This way it has a name that makes sense even outside of the context of prometheus, and we'll be able to re-use it.

What do you think?

Reply:
Indeed, the field name should simply be `Namespace`, the fact that the downstream logic uses this field as a human-readable-id does not need to bleed into the data structure.

BTW, http_check (another check that might have a ton of instances / host), has a mandatory `name` instance field. It might be interesting to also look it up

Reply:
changed it to name, to parse `name` and fallback to `namespace`.
I cannot add the parsing in the method you mentioned @hkaj because I need to call it in BuildID().